docs/traffic-policy/concepts/expressions.mdx

```diff
@@ -0,0 +1,323 @@
+---
+sidebar_position: 3
+title: Expressions
+pagination_previous: traffic-policy/concepts/phase-rules
+pagination_next: traffic-policy/concepts/actions
+---
+
+# Expressions
+
+Expressions are a powerful way to define the conditions under which specific actions in a [traffic policy](/traffic-policy/concepts/rules) are triggered. Expressions are written in the [Common Expression Language (CEL)](https://github.com/google/cel-spec) and allow you to evaluate various traffic attributes, such as headers, paths, IP addresses, and more.
+
+Each expression must evaluate to `true` for the corresponding actions in a rule to be executed. If a rule contains no expressions, it will always evaluate to `true`, causing its actions to be applied to all matching traffic.
+
+## Available Elements
+
+Expressions in traffic policies can reference a variety of elements, including:
+
+- **[Traffic Policy Variables](/traffic-policy/variables)**: These allow you to access metadata about the connection or request, such as certificate details, headers, client IP, geo-location and more.
+- **[Traffic Policy Macros](/traffic-policy/macros)**: Macros simplify complex conditions, enabling reusable patterns across expressions and policies.
+- **[Action Variables](/traffic-policy/actions)**: Some actions may provide additional variables, enabling deeper customization based on specific action outcomes.
+
+## Use-Cases
+
+Expressions give you the flexibility to:
+
+- **Filter traffic**: Evaluate headers, IP addresses, or other attributes to decide if traffic should match a given rule.
+- **Route traffic**: Forward traffic based on factors such as geolocation, protocol, or path.
+- **Control flow**: Implement more complex routing decisions based on dynamic conditions, ensuring optimal traffic handling.
+
+## Common Examples
+
+### Match Based on Header
+
+```yaml
+expressions:
+  - req.headers["authorization"][0] == "Bearer my-token"
+```
+
+### Match Requests to a Specific Path
+
+```yaml
+expressions:
+  - req.path.startsWith("/api/v1/")
+```
+
+### Match by Country
+
+```yaml
+expressions:
+  - conn.geo.country_code == "US"
+```
+
+## Writing Expressions
+
+### Basic Syntax
+
+Expressions are written in CEL and are similar to expressions in languages like C, JavaScript, or Python. They can include variables, literals, and operators.
+
+### Variables
+
+Variables refer to values. For example, `identity` might refer to a user object.
+
+```go
+expressions:
+ - identity.age >= 18 // Checks if the user's age is 18 or more.
```

Comment:
why not use something that works in our policy system as an example here?

Reply:
This was copy pasted. Can update later.